arch/riscv/Kconfig

```diff
@@ -100,6 +100,17 @@ config RISCV_SOC_HAS_CUSTOM_SYS_IO
 	  the RISC-V SoC needs to do something different and more than reading and
 	  writing the registers.
 
+config RISCV_CUSTOM_INTERRUPT_CONTROLLER
+	bool "Custom interrupt controller"
+	depends on !RISCV_HAS_CLIC && !PLIC
```

Comment:
why `PLIC` and not `RISC_HAS_PLIC`?

Reply:
PLIC driver has been built because of CONFIG_PLIC config even with RISCV_HAS_PLIC=n. Later I found that the dependency comes from device tree node, so the right way is to have a clean device tree. 

Let's change it back to RISC_HAS_PLIC, it's even better if the user will become aware that something is not entirely right with his device tree.